uds/database/data_record/text_data_record.py

```diff
@@ -0,0 +1,73 @@
+"""Definition of TextDataRecord which is class for encode and decode values of data records."""
+
+__all__ = ["TextTableDataRecord"]
+
+from typing import Dict, Optional
+
+from .abstract_data_record import AbstractDataRecord, DataRecordPhysicalValueAlias, DecodedDataRecord
+from .raw_data_record import RawDataRecord
+
+
+class TextTableDataRecord(RawDataRecord, AbstractDataRecord):
+    """Implementation for Text Data Record."""
+
+    def __init__(self, name: str, length: int, mapping: Optional[Dict[int, str]] = None) -> None:
+        """
+        Initialize Text Data Record.
+
+        :param name: Name to assign to this Data Record.
+
+        :raise TypeError: Provided value of name is not str type.
+        """
+        super().__init__(name, length)
+        self.length = length
+        self.mapping = mapping
+
+    @property
+    def mapping(self):
+        """Get mapping dict."""
+        return self.__mapping
+
+    @mapping.setter
+    def mapping(self, mapping: dict) -> None:
+        """
+        Set the mapping.
+
+        :param mapping: dict contains mapping.
+        """
+        self.__mapping = mapping
+        self.__reversed_mapping = {v: k for k, v in self.__mapping.items()}
+
+    @property
+    def reversed_mapping(self):
+        """Get reversed mapping dict."""
+        return self.__reversed_mapping
+
+    def decode(self, raw_value: int) -> DecodedDataRecord:  # noqa: F841
+        """
+        Decode physical value for provided raw value.
+
+        :param raw_value: Raw (bit) value of Data Record.
+
+        :return: Dictionary with physical value for this Data Record.
+        """
+        if raw_value in self.mapping:
+            physical_value = self.mapping[raw_value]
+            return DecodedDataRecord(name=self.name, raw_value=physical_value, physical_value=raw_value)
```

Comment:
you mixed the parameters, shall be:
`return DecodedDataRecord(name=self.name, raw_value=raw_value, physical_value=physical_value)`